test/types/lean.test.ts

```diff
@@ -237,3 +237,90 @@ async function gh15057() {
     console.log(item);
   };
 }
+
+async function gh15122() {
+  interface IChild {
+    _id: Types.ObjectId;
+    name: string;
+  }
+
+  type ChildDocumentOverrides = {};
+
+  interface IChildVirtuals {
+    id: string;
+  }
+
+  type ChildInstance = HydratedDocument<
+    IChild,
+    ChildDocumentOverrides & IChildVirtuals
+  >;
+
+  type ChildModelType = Model<
+    IChild,
+    {},
+    ChildDocumentOverrides,
+    IChildVirtuals,
+    ChildInstance
+  >;
+
+
+  interface IParent {
+    _id: Types.ObjectId;
+    name: string;
+    surname: string;
+    child: PopulatedDoc<Document<Types.ObjectId> & IChild>;
+  }
+
+  type ParentDocumentOverrides = {};
+
+  interface IParentVirtuals {
+    id: string;
+    fullName: string;
+  }
+
+  type ParentInstance = HydratedDocument<
+    IParent,
+    ParentDocumentOverrides & IParentVirtuals
+  >;
+
+  type ParentModelType = Model<
+    IParent,
+    {},
+    ParentDocumentOverrides,
+    IParentVirtuals,
+    ParentInstance
+  >;
+
+  const parentSchema = new Schema<IParent, ParentModelType>(
+    {
+      name: {
+        type: String,
+        required: true,
+        trim: true
+      },
+      surname: {
+        type: String,
+        required: true,
+        trim: true
+      },
+      child: {
+        type: 'ObjectId',
+        ref: 'Child',
+        required: true
+      }
+    },
+    { optimisticConcurrency: true }
```

Comment:
```suggestion
    }
```

i think this can be omitted as it is not relevant to types, right?

Reply:
Correct, `optimisticConcurrency: true` has no impact on this test.